src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -96,26 +97,29 @@ public async Task SpecializeAsync()
             _shutdownStandbyWorkerChannels();
         }
 
-        public bool ShutdownChannelIfExists(string language, string workerId)
+        public async Task<bool> ShutdownChannelIfExists(string language, string workerId)
         {
             if (string.IsNullOrEmpty(language))
             {
                 throw new ArgumentNullException(nameof(language));
             }
-            if (_workerChannels.TryGetValue(language, out List<ILanguageWorkerChannel> languageWorkerChannels))
+            if (_workerChannels.TryGetValue(language, out Dictionary<string, TaskCompletionSource<ILanguageWorkerChannel>> languageWorkerChannels))
             {
-                var channel = languageWorkerChannels.FirstOrDefault(ch => ch.Id == workerId);
-                if (channel != null)
+                if (languageWorkerChannels.TryGetValue(workerId, out TaskCompletionSource<ILanguageWorkerChannel> value))
                 {
-                    (channel as IDisposable)?.Dispose();
-                    languageWorkerChannels.Remove(channel);
-                    return true;
+                    ILanguageWorkerChannel channel = await value.Task;
+                    if (channel != null)
+                    {
+                        (channel as IDisposable)?.Dispose();
+                        languageWorkerChannels.Remove(workerId);
+                        return true;
+                    }
                 }
             }
             return false;
         }
 
-        internal void ScheduleShutdownStandbyChannels()
+        internal async void ScheduleShutdownStandbyChannels()
```

Comment:
>async void [](start = 17, length = 10)

async Task?

Reply:
This is an action and ok to be async void. Fixed rest of them!